File: Python/Math-Array-String/46.py
# 2129. Capitalize the Title


class Solution:
    def capitalizeTitle(self, title: str) -> str:
        # Split the title into words
        words = title.split()
        
        # Process each word based on its length
        capitalized_words = []
        for word in words:
            if len(word) <= 2:
                # If the word length is 1 or 2, make it lowercase
                capitalized_words.append(word.lower())
            else:
                # If the word length is greater than 2, capitalize the first letter and lowercase the rest
                capitalized_words.append(word[0].upper() + word[1:].lower())
        
        # Join the processed words back into a single string
        return " ".join(capitalized_words)
    

sol = Solution()
output = sol.capitalizeTitle("capiTalIze tHe titLe") 
print(output)

File: JavaScript/Math-Array-String/46.js
// 2129. Capitalize the Title


/**
 * @param {string} title
 * @return {string}
 */
var capitalizeTitle = function(title) {
    // Split the title into words
    let words = title.split(" ");
    
    // Process each word based on its length
    let capitalizedWords = words.map(word => {
        if (word.length <= 2) {
            // If the word length is 1 or 2, convert it to lowercase
            return word.toLowerCase();
        } else {
            // If the word length is greater than 2, capitalize the first letter and lowercase the rest
            return word[0].toUpperCase() + word.slice(1).toLowerCase();
        }
    });
    
    // Join the processed words back into a single string
    return capitalizedWords.join(" ");
};

// Example usage
console.log(capitalizeTitle("capiTalIze tHe titLe")); // Output: "Capitalize The Title"
console.log(capitalizeTitle("First leTTeR of EACH Word")); // Output: "First Letter of Each Word"
console.log(capitalizeTitle("i lOve leetcode")); // Output: "i Love Leetcode"
